README.md

```diff
@@ -1,6 +1,100 @@
+* project analysis
+* network proxying
 
+# Installation
+
+*roku-dev-cli* can be installed from the pypi repository using pip.  
+
+```shell
+pip install roku-dev-cli
+```
+
+# Loading a Build
+
+To load a build on the Roku, first make sure the device is on the same network and is in developer mode ([how to enter developer mode](https://sdkdocs.roku.com/display/sdkdoc/Loading+and+Running+Your+Application#LoadingandRunningYourApplication-EnablingDevelopmentModeonyourbox)).  
+
+In your terminal, go to the base directory of your Roku project where the *manifest* file exists.  Then run the following command:
+
+```shell
+roku <IP_ADDRESS>
+```
+This will zip up the current directory and send it to the device.  If the app is successfully installed the tool will output the application's log to your console as it executes.  
+
+The script will do some basic checking of the project making sure all the component files are valid XML.  
+
+# Loading a build from an existing zip file
+If you already have a zip file you want to load, you can use the zip_file flag instead of zipping the current directory:
+
+```shell
+roku --zip_file <ZIP_FILE_PATH> <IP_ADDRESS>
+```
+
+This will install the zip file and start logging the application output.  
+
+# Analysis
+TODO: document analysis tools and graphing
+
+# Network Proxying
+The tool has some support for *mitmproxy*, which allows you to view and control network traffic running through the box.  
+
+The Roku OS currently does not support HTTP proxy forwarding, so users must first route traffic through a computer.  
+
+### Step 0: Install mitmproxy
+
+On mac, run the following command:
+
+```shell
+brew install mitmproxy
+```
+
+### Step 1: Setup routing
+
+#### On Linux:
+```shell
+sudo sysctl -w net.ipv4.ip_forward=1
+sudo iptables -t nat -A PREROUTING -i wlan1 -p tcp --dport 443 -j REDIRECT --to-port 8080
+sudo iptables -t nat -A PREROUTING -i wlan1 -p tcp --dport 80 -j REDIRECT --to-port 8080
+```
+Replace *wlan1* with the wifi interface the Roku device is connected to.  
+
+#### On Mac:
+Enable port forwarding:
+```shell
+sudo sysctl -w net.inet.ip.forwarding=1
+```
+
+Update pf config.  It should look something like this...
+```shell
+...
+scrub-anchor "com.apple/*"
+nat-anchor "com.apple/*"
+rdr-anchor "com.apple/*"
+# Port forwarding for mitmproxy
+rdr pass on bridge100 inet proto tcp from any to any port 80 -> 127.0.0.1 port 8080
+rdr pass on bridge100 inet proto tcp from any to any port 443 -> 127.0.0.1 port 8080
+dummynet-anchor "com.apple/*"
+anchor "com.apple/*"
+...
+```
+Notice that the forwarding rules were inserted between some other existing rules, if you append them to the end of the file you will get some warnings. 
+
+Finally run the following commands.  The first will check file syntax while the second one will restart the service loading the new rules.
+```shell
+sudo pfctl -v -n -f /etc/pf.conf
+sudo pfctl -ef /etc/pf.conf
+```
+
+### Step 2: Start the proxy
```

Comment:
Despite the name, this step doesn’t mention how to start `mitmproxy` for use with the `roku` CLI. If I recall correctly, it requires something like `—transparent` and it has to run on port `8888`, right? If you get a minute, it’d be _super_ helpful to have that documented.

Mostly because I can’t remember how to do it 🙃

Reply:
This is now started on by the tool itself including the weird parameters.  Yay!  mitmproxy defaults to using 8080, which the instructions above mention for routing.  

I do think there's something lacking in this area so I'd like to watch a newcomer go through the steps and see which parts are confusing.